File: STC/git_testing/routines.c
#include <math.h>
void isqrt(float *a, int n){
        int i;
//                      Vector inverse square
//                      Someone advised to 
//                      make two separate loops.

        // combined both loops
        for (i=0; i<n; i++){

            a[i] = sqrt(a[i]);
            a[i] = 1.0e0/a[i];
        }

//        for (i=0; i<n; i++) a[i] = 1.0e0/a[i];
}

File: STC/git_test/routines.f90
subroutine isqrt(a, n)
    integer :: n,i
    real    :: a(n)
!                   Vector inverse square
!                   Someone advised to
!                   making two separate loops.
!           combined both loops
    do i=1,n;
        a(i) = sqrt(a(i)); 
        a(i) = 1.0e0/a(i);
    end do
end subroutine
